File: backend/module/Radio/src/Radio/Entity/Bookmark.php
<?php

namespace Radio\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ORM\Table(name="bookmark")
 * */
class Bookmark
{

    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue
     * */
    protected $id;
    /**
     * @ORM\Column(type="datetime")
     * */
    protected $start;
    /**
     * @ORM\Column(type="datetime")
     * */
    protected $end;
    /**
     * @ORM\ManyToOne(targetEntity = "Show")
     * @ORM\JoinColumn(name = "radioshow_id", referencedColumnName = "id")
     */
    protected $show;
    /**
     * @ORM\ManyToOne(targetEntity = "Episode")
     * @ORM\JoinColumn(name = "episode_id", referencedColumnName = "id")
     */
    protected $episode;

    /**
     * @ORM\ManyToOne(targetEntity = "User")
     * @ORM\JoinColumn(name = "user_id", referencedColumnName = "id")
     */
    protected $author;
    /**
     * @ORM\Column(type="datetime")
     * */
    protected $modified;
    /**
     * @ORM\Column(type="string",length=160)
     * */
    protected $title;
    /**
     * @ORM\Column(type="text")
     * */
    protected $content;

    /**
     * @return mixed
     */
    public function getAuthor()
    {
        return $this->author;
    }

    /**
     * @param mixed $author
     */
    public function setAuthor($author)
    {
        $this->author = $author;
    }

    /**
     * @return mixed
     */
    public function getContent()
    {
        return $this->content;
    }

    /**
     * @param mixed $content
     */
    public function setContent($content)
    {
        $this->content = $content;
    }

    /**
     * @return mixed
     */
    public function getEpisode()
    {
        return $this->episode;
    }

    /**
     * @param mixed $episode
     */
    public function setEpisode($episode)
    {
        $this->episode = $episode;
    }


    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getModified()
    {
        return $this->modified;
    }

    /**
     * @param mixed $modified
     */
    public function setModified($modified)
    {
        $this->modified = $modified;
    }

    /**
     * @return mixed
     */
    public function getShow()
    {
        return $this->show;
    }

    /**
     * @param mixed $show
     */
    public function setShow($show)
    {
        $this->show = $show;
    }

    /**
     * @return mixed
     */
    public function getTitle()
    {
        return $this->title;
    }

    /**
     * @param mixed $title
     */
    public function setTitle($title)
    {
        $this->title = $title;
    }

    /**
     * @return mixed
     */
    public function getEnd()
    {
        return $this->end;
    }

    /**
     * @param mixed $end
     */
    public function setEnd($end)
    {
        $this->end = $end;
    }

    /**
     * @return mixed
     */
    public function getStart()
    {
        return $this->start;
    }

    /**
     * @param mixed $start
     */
    public function setStart($start)
    {
        $this->start = $start;
    }


}

?>

File: util/model/src/main/java/hu/radio/tilos/model/Bookmark.java
package hu.radio.tilos.model;

import hu.radio.tilos.model.type.MixCategory;
import hu.radio.tilos.model.type.MixType;

import javax.persistence.*;
import java.util.Date;

@Entity()
@Table(name = "bookmark")
public class Bookmark {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @Basic
    @Column(length = 160)
    private String title;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "start")
    private Date realFrom;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "end")
    private Date realTo;

    @ManyToOne()
    @JoinColumn(name = "radioshow_id", referencedColumnName = "id")
    private Show show;

    @ManyToOne()
    @JoinColumn(name = "episode_id", referencedColumnName = "id")
    private Episode episode;

    @Basic
    @Column(name = "full_episode", nullable = false, columnDefinition = "TINYINT(1)")
    private boolean fullEpisode = false;

    @Basic
    @Column(nullable = false, columnDefinition = "TINYINT(1)")
    private boolean selected = false;

    @Basic
    @Column()
    private int karma;

    @ManyToOne()
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User author;

    public boolean isFullEpisode() {
        return fullEpisode;
    }

    public void setFullEpisode(boolean fullEpisode) {
        this.fullEpisode = fullEpisode;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    public int getKarma() {
        return karma;
    }

    public void setKarma(int karma) {
        this.karma = karma;
    }

    public Show getShow() {
        return show;
    }

    public void setShow(Show show) {
        this.show = show;
    }

    public Episode getEpisode() {
        return episode;
    }

    public void setEpisode(Episode episode) {
        this.episode = episode;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Date getRealFrom() {
        return realFrom;
    }

    public void setRealFrom(Date realFrom) {
        this.realFrom = realFrom;
    }

    public Date getRealTo() {
        return realTo;
    }

    public void setRealTo(Date realTo) {
        this.realTo = realTo;
    }

    public User getAuthor() {
        return author;
    }

    public void setAuthor(User author) {
        this.author = author;
    }
}
